### PR TITLE
Vue & React app generator updates

### PR DESCRIPTION
**Pull request by Sahana-Srinivasan**
Used blockstack-todos Vue packaging and modified Profile/Signed components from estermer's [PR 15](https://github.com/blockstack/blockstack-app-generator/pull/15) to make a Hello, Blockstack app generator for Vue. Minifies the packaging of the React generator and updates the style of all four generators to match blockstack.org.

### CODE REVIEW

**Comment by Sahana-Srinivasan** [comment]
Updated the Profile/SignIn components to reflect new versions of Blockstack with slightly different sign-in process. Combined with the updated packaging of blockstack-todos for an updated "Hello, Blockstack" Vue app generator in [this pull request](https://github.com/blockstack/blockstack-app-generator/pull/54).

**Comment by hstove** [comment]
The git diff is hard to parse. It seems like some React and Webpack code is also altered? Is this PR built on top of a different PR that isn't merged to master, yet?

**Comment by Sahana-Srinivasan** [comment]
> The git diff is hard to parse. It seems like some React and Webpack code is also altered? Is this PR built on top of a different PR that isn't merged to master, yet?

This encapsulates the React generator updates made in PR 50, yeah. The general app generator/Webpack generator received minor updates just to make the Hello Blockstack page blue and look more similar to blockstack.org. Updated the PR comment.

**Comment by zone117x** [comment]
Looks like this needs the master branch merged in (or rebased on top of the master branch)

**Comment by Sahana-Srinivasan** [comment]
> The git diff is hard to parse. It seems like some React and Webpack code is also altered? Is this PR built on top of a different PR that isn't merged to master, yet?

@hstove Yeah, it modifies the blockstack logos/color for all the generators, improves the packaging for the React generator, creates a more robust test script, and adds the Vue generator. Also, Matt and I went through the merge conflicts so the diff should be more coherent now.

**Comment by Sahana-Srinivasan** [comment]
@zone117x Does this look good to go?

File: app/templates/_package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "browserify": "browserify requires.js -o public/bundle.js",
-    "start": "npm run browserify && node server.js"
+    "start": "npm run browserify && node server.js",
+    "test": "echo works"
```
Comment (zone117x) [inline app/templates/_package.json]: This should probably be changed to something like `echo "TODO: write tests"`

File: vue/templates/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+version: '2'
```
Comment (zone117x) [inline vue/templates/docker-compose.yaml]: Is this file being used anywhere or still work?
The docker images appear to be kinda old, and I'm not seeing any references of this file.

File: vue/templates/browserslistrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+> 1%
```
Comment (zone117x) [inline vue/templates/browserslistrc]: Should this be moved into the `_package.json` to be consistent the other generators? There's also another file `.postcssrc.js` in this directory that states `// to edit target browsers: use "browserlist" field in package.json`.

(I know this this Vue template code is from another contributor's older PR)

File: vue/templates/_package.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/blockstack/blockstack-todos"
+  },
+  "private": true,
+  "scripts": {
+    "start": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "lint": "vue-cli-service lint",
+    "test:unit": "vue-cli-service test:unit --coverage --coverageReporters=lcov",
```
Comment (zone117x) [inline vue/templates/_package.json]: When running `npm run test:unit` I'm getting an error:
```
> vue-cli-service test:unit --coverage --coverageReporters=lcov

Error: Could not resolve a module for a custom reporter.
  Module name: jest-junit
    at options.reporters.reporters.map.reporterConfig (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest-config/build/normalize.js:419:15)
    at Array.map (<anonymous>)
    at normalizeReporters (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest-config/build/normalize.js:397:33)
    at normalize (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest-config/build/normalize.js:494:5)
    at readConfig (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest-config/build/index.js:184:68)
    at getConfigs (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest/node_modules/jest-cli/build/cli/index.js:473:77)
    at /Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest/node_modules/jest-cli/build/cli/index.js:251:23
    at Generator.next (<anonymous>)
    at step (/Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest/node_modules/jest-cli/build/cli/index.js:206:30)
    at /Users/matt/Projects/blockstack-app-generator/.vue-gen-test/node_modules/jest/node_modules/jest-cli/build/cli/index.js:225:14
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bs-todo@6.0.1 test:unit: `vue-cli-service test:unit --coverage --coverageReporters=lcov`
npm ERR! Exit status 1
```

I'm not really familiar with Vue or what code/tests this is trying to exercise -- should we fix this or remove it?

File: react/templates/_package.json [hunk 5]
```diff
@@ -2,41 +2,45 @@
...
-    "react-dom": "^15.5.4"
+    "blockstack": "^19.2.1",
+    "react": "^16.8.6",
+    "react-dom": "^16.8.6",
+    "react-router-dom": "^4.3.1",
+    "react-scripts": "3.0.1"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
```
Comment (zone117x) [inline react/templates/_package.json:23]: Getting some minor linting warnings when building:
```
Compiled with warnings.

./src/App.js
  Line 14:  Useless constructor  no-useless-constructor

./src/index.js
  Line 1:  'Component' is defined but never used  no-unused-vars
  Line 7:   'bootstrap' is defined but never used  no-unused-vars
  Line 8:   'style' is defined but never used      no-unused-vars

./src/Signin.js
  Line 4:  Useless constructor  no-useless-constructor
```

File: app/index.js [hunk 6]
```diff
@@ -4,6 +4,13 @@ var chalk = require('chalk');
 var yosay = require('yosay');
```
Comment (zone117x) [inline app/index.js:4]: I'm tempted to say we nuke this template and replace it with either the `react` or `webpack` template. Both of those templates generate a more concise, useful and standard app scaffold than this one. Probably something that should be done in a later PR though.
@moxiegirl @yknl thoughts?

**Review by zone117x — CHANGES_REQUESTED** [review]
Tested all the generated apps, and all working for me on Chrome. The new theming looks great!
Left a few comments & questions about minor stuff. Other than addressing those, this PR looks good to me!

File: react/templates/_package.json [hunk 7]
```diff
@@ -2,41 +2,45 @@
...
-    "url-loader": "^0.5.8",
-    "webpack": "^2.3.3",
-    "webpack-dev-server": "^2.4.2"
-  },
   "dependencies": {
-    "blockstack": "^19.1.0",
-    "bootstrap": "^3.3.7",
-    "react": "^15.5.4",
-    "react-dom": "^15.5.4"
+    "blockstack": "^19.2.1",
```
Comment (yknl) [inline react/templates/_package.json]: We can update this to 19.2.2 now.

**Review by yknl — CHANGES_REQUESTED** [review]
Nice work @Sahana-Srinivasan. The updated UI looks great. Just a minor change in addition to what Matt added.

**Review by yknl — APPROVED** [review]
LGTM

**Review by zone117x — APPROVED** [review]
LGTM. Great job @Sahana-Srinivasan !
